components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -314,16 +314,14 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
         try {
             isValidGrant = authzGrantHandler.validateGrant(tokReqMsgCtx);
         } catch (IdentityOAuth2Exception e) {
-            if (log.isDebugEnabled()) {
-                log.debug("Error occurred while validating grant", e);
-            }
             if (e.getErrorCode() != null) {
                 errorCode = e.getErrorCode();
             }
             error = e.getMessage();
             if (e.getErrorCode() != null) {
                 errorCode = e.getErrorCode();
             }
+            log.error("Error occurred while validating grant: " +  error);
```

Comment:
Shouldn't we log the `e` (exception) here instead of the `error` String

Reply:
We decided to add the error message instead of logging the exception. 

Related PR review: https://github.com/wso2-support/identity-inbound-auth-oauth/pull/1292#discussion_r1050555558